### PR TITLE
Borrow&requests part2

### PR DESCRIPTION
**Pull request by IsaakC**
--- new request ---

Created abstract class DetailView to refactor code, it implements parts that all DetailViews need

Cleaned onClick in BookListAdapter by creating method that returns View.OnClickListeners for a given fragment

Adapted DetailViews to work with new abstract class

--- old request ---
Added Booklists to Request tab, one for request one for borrow

to reuse code I created a book_detail_common xml for the stuff all detailed views use

I changed the BookListAdapter to take an argument from the fragment to determine which onClick activity to do

### CODE REVIEW

**Comment by n00b4ever** [comment]
needs book owner username and it will all come together

**Review by n00b4ever — COMMENTED** [review]
need owner user name in book and it will all come together

merge all lines, delete this?
image

**Comment by IsaakC** [comment]
The DetailViews have spots for Borrower / Owner usernames. They are not finalized yet because they have status dependent buttons, prompts, etc. 

In this PR I want to get the BookListAdapter and DetailView skeletons in so I can start finalizing the DetailView with all the new changes on the main branch

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java [hunk 1]
```diff
@@ -51,23 +66,86 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        requestsRecyclerView = (RecyclerView) view.findViewById(R.id.requests_recycler_view);
+        requestsRecyclerView.setHasFixedSize(true);
+
+        requestsRecyclerLayoutManager = new LinearLayoutManager(this.getContext());
+        requestsRecyclerView.setLayoutManager(requestsRecyclerLayoutManager);
+
+        requestsRecyclerAdapter = new BookListAdapter(this.getContext(), options, R.id.requests);
+        requestsRecyclerView.setAdapter(requestsRecyclerAdapter);
+
+        Query query2 = rootRef.collection(getString(R.string.books_collection)).orderBy("title");
```
Comment (huddy987) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java:92]: Why are we doing 2 queries?

**Review by huddy987 — CHANGES_REQUESTED** [review]
Screenshots?

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java (hunk 1 above)
Comment (IsaakC) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java:92]: there are two lists, own for requests on your books, other for your requests

**Comment by IsaakC** [comment]
image
image
image
image

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java (hunk 1 above)
Comment (huddy987) [reply app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java:92]: Ok these queries will need to be renamed but I think its fine for now

**Review by huddy987 — APPROVED** [review]

**Review by n00b4ever — COMMENTED** [review]
read it last night

**Review by n00b4ever — APPROVED** [review]

**Pull request merged**